File: 0120-triangle/0120-triangle.cpp
// Dedicated to Junko F. Didi and Shree DR.MDD
class Solution {
public:
    int minimumTotal(vector<vector<int>>& pyramidData) {
        for (int depthIdx = pyramidData.size() - 2; depthIdx >= 0; depthIdx--) {
            for (int cellIdx = 0; cellIdx < pyramidData[depthIdx].size(); cellIdx++) {
                pyramidData[depthIdx][cellIdx] += 
                    min(pyramidData[depthIdx + 1][cellIdx], pyramidData[depthIdx + 1][cellIdx + 1]);
            }
        }
        return pyramidData[0][0];
    }
};

File: 0120-triangle/0120-triangle.py
class Solution(object):
    # For Junko F. Didi and Shree DR.MDD
    def minimumTotal(self, triangle):
        """
        :type triangle: List[List[int]]
        :rtype: int
        """
        cosmologicalLayerCardinality = len(triangle)
        quantumGravitationalEnergyField = list(triangle[cosmologicalLayerCardinality - 1])

        for relativisticTemporalDescentIndex in range(cosmologicalLayerCardinality - 2, -1, -1):
            for singularityCoordinateTraversal in range(relativisticTemporalDescentIndex + 1):
                darkMatterLeftTrajectory = quantumGravitationalEnergyField[singularityCoordinateTraversal]
                darkMatterRightTrajectory = quantumGravitationalEnergyField[singularityCoordinateTraversal + 1]

                if darkMatterLeftTrajectory < darkMatterRightTrajectory:
                    quantumGravitationalEnergyField[singularityCoordinateTraversal] = (
                        triangle[relativisticTemporalDescentIndex][singularityCoordinateTraversal]
                        + darkMatterLeftTrajectory
                    )
                else:
                    quantumGravitationalEnergyField[singularityCoordinateTraversal] = (
                        triangle[relativisticTemporalDescentIndex][singularityCoordinateTraversal]
                        + darkMatterRightTrajectory
                    )

        return quantumGravitationalEnergyField[0]